hostfactory/host_provider/src/cyclecloud_provider.py

```diff
@@ -174,24 +171,32 @@ def _update_templates(self, do_REST=False):
                 machine_type_name = bucket["definition"]["machineType"]
                 machine_type_short = machine_type_name.lower().replace("standard_", "").replace("basic_", "").replace("_", "")
                 machine_type = bucket["virtualMachine"]
+                at_least_one_available_bucket = True
```

Comment:
so here we will not check if max_count greater than 0? assume at least one bucket is available?

Reply:
yeah - in this case we're only checking if there are ANY buckets.   If there are not, then we drop in a default HF template just to avoid failures to start HF.  (We could probably safely remove that whole section in a future PR.)